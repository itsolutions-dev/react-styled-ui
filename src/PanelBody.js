import styled, { css } from './styledComponent';
import { getBackgroundColor } from './utils/';

const PanelBody = styled({
  component: 'div',
  style: css`
    font-size: 16px;
    text-indent: 20px;
    padding: 18px;
    ${(props) => {
      const color = props.borderColor !== undefined ? props.borderColor : getBackgroundColor(props);
      return `
        border-left: 1px solid ${color};
        border-right: 1px solid ${color};
      `;
    }}
  `,
});

PanelBody.defaultProps = {
  primary: true,
};

export default PanelBody;
