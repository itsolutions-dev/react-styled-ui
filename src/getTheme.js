import defaultTheme from './theme';
import deepMerge from './utils/deepMerge';

const getTheme = deepMerge.bind(null, defaultTheme);

export default getTheme;
