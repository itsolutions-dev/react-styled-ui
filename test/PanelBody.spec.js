import React from 'react';
import ReactTestRenderer from 'react-test-renderer';
import 'styled-components-test-utils/lib/jest';
import getTheme from '../src/getTheme';
import PanelBody from '../src/PanelBody';

const theme = getTheme({
  radius: '8px',
  palette: {
    primary: 'blue',
  },
});

describe('PanelBody', () => {
  test('should render a PanelBody', () => {
    const component = ReactTestRenderer.create(<PanelBody theme={theme} />);
    expect(component).toBeDefined();
    expect(component.toJSON()).toMatchSnapshot();
  });

  test('should have a div tag', () => {
    const component = ReactTestRenderer.create(<PanelBody theme={theme} />).toJSON();
    expect(component.type).toEqual('div');
  });

  test('should have the given border color', () => {
    const component = ReactTestRenderer.create(<PanelBody borderColor="green" theme={theme} />);
    expect(component).toHaveStyleRule('border-left', '1px solid green');
    expect(component).toHaveStyleRule('border-right', '1px solid green');
  });

  test('should be primary by default', () => {
    const component = ReactTestRenderer.create(<PanelBody theme={theme} />);
    expect(component).toHaveStyleRule('border-left', '1px solid blue');
    expect(component).toHaveStyleRule('border-right', '1px solid blue');
  });
});
