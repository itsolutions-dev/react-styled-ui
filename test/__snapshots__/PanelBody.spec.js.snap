// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PanelBody should render a PanelBody 1`] = `
.c0 {
  font-size: 16px;
  text-indent: 20px;
  padding: 18px;
  border-left: 1px solid blue;
  border-right: 1px solid blue;
}

<div
  className="c0"
/>
`;
